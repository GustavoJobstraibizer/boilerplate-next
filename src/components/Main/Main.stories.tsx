import { Meta, Story } from '@storybook/react'
import Main from '.'

export default {
  title: 'Main',
  component: Main,
  args: {
    title: 'React AvanÃ§ado',
    description: 'Typescript, ReactJS, NextJS e Styled Components'
  }
} as Meta

export const Basic: Story = (args) => <Main {...args} />
