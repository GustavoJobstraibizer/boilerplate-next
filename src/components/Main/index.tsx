const Main = () => {
  return (
    <div data-testid="main">
      <h1>Hello World</h1>

      <span>teste</span>
    </div>
  )
}

export default Main
