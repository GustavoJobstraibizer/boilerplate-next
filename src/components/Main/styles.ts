import Image from 'next/image'
import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  display: flex;
  text-align: center;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  padding: 3rem;
`
export const Logo = styled(Image)`
  width: 25rem;
  margin-bottom: 2rem;
`

export const Title = styled.h1`
  font-size: 2.5rem;
`

export const Description = styled.h2`
  font-size: 2rem;
  font-weight: 400;
`

export const Illustration = styled(Image)`
  margin: 3rem;
  width: min(30rem, 100%);
`
