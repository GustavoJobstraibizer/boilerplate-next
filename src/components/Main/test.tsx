import { render, screen } from '@testing-library/react'
import Main from '.'

describe('Main', () => {
  it('should render Main component', () => {
    const { container } = render(<Main />)

    expect(screen.getByTestId('main')).toBeInTheDocument()
    expect(screen.getByText(/Hello world/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
