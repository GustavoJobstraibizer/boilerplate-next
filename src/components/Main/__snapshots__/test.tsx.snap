// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render Main component 1`] = `
<div
  data-testid="main"
>
  <h1>
    Hello World
  </h1>
  <span>
    teste
  </span>
</div>
`
